pandas/core/arrays/period.py

```diff
@@ -873,15 +782,60 @@ def _add_delta(self, other):
                                         .format(cls=type(self).__name__,
                                                 freqstr=self.freqstr))
 
-        # TODO: standardize across datetimelike subclasses whether to return
-        #  i8 view or _shallow_copy
-        if isinstance(other, (Tick, timedelta, np.timedelta64)):
-            return self._add_delta_td(other)
-        elif is_timedelta64_dtype(other):
-            # ndarray[timedelta64] or TimedeltaArray/index
-            return self._add_delta_tdi(other)
-        else:  # pragma: no cover
-            raise TypeError(type(other).__name__)
+        new_ordinals = super(PeriodArray, self)._add_delta(other)
+        return type(self)(new_ordinals, freq=self.freq)
+
+    def _check_timedeltalike_freq_compat(self, other):
```

Comment:
this is simply moved? (or is also the code changed?)

Reply:
This is a new function that is similar-but-not-identical to _maybe_convert_timedelta.  The latter is no longer needed by the Array, so is moved back to the Index.

Reply:
Sorry, I was not talking about `_maybe_convert_timedelta`, but the existing `_check_timedeltalike_freq_compat`, which, according to the diff here already existed in this file, but is simply moved (or also adapted?). 
If there is a good reason to move it within the file (eg to a more logical order), that's fine, otherwise I would keep it where it was above to have a smaller diff.

Reply:
The PeriodArray PR moved the arithmetic functions, which this belongs adjacent to.  This moves it to keep it with the arithmetic functions.

Reply:
OK, that's a good reason!